### PR TITLE
feat(DSM-2126): use remote adb url if present for secureadb connection

### PR DESCRIPTION
**Pull request by nazeel-esper**
This PR adds support to TCP relay server connection using URL. 
For backward compatibility: If url doesn’t exist, connect to provided ip.

### CODE REVIEW

File: esper/ext/remoteadb_api.py [hunk 1]
```diff
@@ -104,6 +105,15 @@ def fetch_relay_endpoint(environment: str,
...
             host = remoteadb_session.get("ip")
             port = remoteadb_session.get("client_port")
+            remoteadb_host = remoteadb_session.get("remoteadb_host")
+
+            if remoteadb_host:
+                try:
+                    ip_address = socket.gethostbyname(remoteadb_host)
+                    host = ip_address
+                except socket.error as e:
+                    log.debug(f"[remoteadb-connect] Could not resolve ip address of '{remoteadb_host}', falling back to {host}. Error was: {e}")
```
Comment (kimt-esper) [inline esper/ext/remoteadb_api.py:115]: We are not falling back to `host` implementation right? Could you edit the log accordingly?

Comment (nazeel-esper) [reply esper/ext/remoteadb_api.py]: Resolved. Also, used `error` instead of `debug`.

**Review by kimt-esper — APPROVED** [review]

File: esper/ext/remoteadb_api.py [hunk 2]
```diff
@@ -104,6 +105,15 @@ def fetch_relay_endpoint(environment: str,
...
             port = remoteadb_session.get("client_port")
+            remoteadb_host = remoteadb_session.get("remoteadb_host")
+
+            if remoteadb_host:
+                try:
+                    ip_address = socket.gethostbyname(remoteadb_host)
+                    host = ip_address
+                except socket.error as e:
+                    log.error(f"[remoteadb-connect] Could not resolve ip address of '{remoteadb_host}'. Error was: {e}")
+                    raise e
```
Comment (abdul-esper) [inline esper/ext/remoteadb_api.py:116]: Raise user identifiable error.

**Comment by abdul-esper** [comment]
Add @jerynmathew also as a reviewer.

**Review by kimt-esper — APPROVED** [review]

**Pull request merged**